dotnet/src/Connectors/Connectors.GoogleVertexAI/Core/Gemini/Common/GeminiChatCompletionClient.cs

```diff
@@ -91,35 +92,48 @@ internal class GeminiChatCompletionClient : GeminiClient, IGeminiChatCompletionC
         }
     }
 
-    private static void ValidateAndPrepareChatHistory(ChatHistory chatHistory)
+    private static void ValidateAndPrepareChatHistory(ref ChatHistory chatHistory)
     {
         Verify.NotNullOrEmpty(chatHistory);
 
-        if (chatHistory.SingleOrDefault(message => message.Role == AuthorRole.System) is { } systemMessage)
+        chatHistory = new ChatHistory(chatHistory);
+
+        if (chatHistory.Where(message => message.Role == AuthorRole.System).ToList() is { Count: > 0 } systemMessages)
         {
-            if (chatHistory.Count == 1)
+            if (chatHistory.Count == systemMessages.Count)
```

Comment:
Confusing in this part, if multiple can become a user message, why having only one gives an exception, to be concise with the behavior, I would expect that single system to become an user message.

Reply:
It doesn't matter is none/one/many system messages in chat history. Exception would only be thrown in case chat history contains only system messages.
System message one or more always is converted to one user message.